### ISSUE TITLE
Multiline Secrets are rendered in console output

### ISSUE THREAD

**Issue by gregorydlogan** [buildbot/buildbot#4793]
Attempting to distribute a PGP key temporarily for code signing purposes, I'm noticing the secret being redacted in the initial step description, but not in the argv line.  Single line secrets are properly redacted, this appears to only affect multiline secrets:

```python
steps.ShellSequence(
        commands=[
            util.ShellArg(
                command=util.Interpolate("echo '%(secret:signing.key)s' > /tmp/key"),
                logfile="send"),
            util.ShellArg(
                command=["gpg", "--import", "/tmp/key"],
                logfile="load"),
            util.ShellArg(
                command=["rm", "-f", "/tmp/key"],
                logfile="cleanup")
        ],
        haltOnFailure=True,
        flunkOnFailure=True,
        name="Load signing key")
```

Yields this in the step log
```bash
echo '<signing.key>' > /tmp/key
 in dir /builds/Develop_Debian_Packaging/build (timeout 1200 secs)
 watching logfiles {}
 argv: b"echo '-----BEGIN PGP PRIVATE KEY BLOCK-----\n\nl...
```

**Comment by tardyp** [comment]
Indeed, this is a problem, as the secret redaction is made as early as possible, when a chunk of log is received:
https://github.com/buildbot/buildbot/blob/434c0556ff023429e5f727ae5df722cb8fd2d8d1/master/buildbot/process/remotecommand.py#L276

If the secret is split between to chunk of data, it may not be redacted.

In your case however, it looks like it is more your secret contains \n, and the secret is printed in repr() mode, so  the comparaison does not match as '\n' are replaced by '\\n'

a workaround would be to also filter out the repr of the secret:

https://github.com/buildbot/buildbot/blob/434c0556ff023429e5f727ae5df722cb8fd2d8d1/master/buildbot/process/properties.py#L207-L209

patched to:
```python
    def useSecret(self, secret_value, secret_name):
        self._used_secrets[secret_value] = "<" + secret_name + ">"
        self._used_secrets[repr(secret_value)] = "<" + secret_name + ">"
```

**Comment by gregorydlogan** [comment]
I worked around my issue by slinging the files via SCP, but I can certainly test locally.  Would the workaround resolve the issue with the log chunking?  Or would the patch you provided likely resolve the issue.  I'm not sure I can reliably test the log chunking :)

**Comment by tardyp** [comment]
I am pretty sure the log that you sent will not be chunked. only logs generated by subprocess commands can be chunked (by the OS io buffering system)

**Comment by PhilippSelenium** [comment]
Is there a reason the patch is not applied?

**Comment by tardyp** [comment]
Somebody has to take time to make a PR, and add unit tests.
